src/mono/mono/metadata/mono-debug.c

```diff
@@ -893,18 +893,16 @@ mono_debug_lookup_locals (MonoMethod *method, mono_bool ignore_pdb)
 		return NULL;
 	}
 
-	if (ignore_pdb)
-		res = mono_debug_symfile_lookup_locals (minfo);
-	else {
-		if (minfo->handle->ppdb) {
-			res = mono_ppdb_lookup_locals (minfo);
-		} else {
-			if (!minfo->handle->symfile || !mono_debug_symfile_is_loaded (minfo->handle->symfile))
-				res = NULL;
-			else
-				res = mono_debug_symfile_lookup_locals (minfo);
-		}
+
+	if (minfo->handle->ppdb) {
+		res = mono_ppdb_lookup_locals (minfo);
+	} else {
+		if (!minfo->handle->symfile || !mono_debug_symfile_is_loaded (minfo->handle->symfile))
+			res = NULL;
+		else
+			res = mono_debug_symfile_lookup_locals (minfo);
```

Comment:
This is handling the case of embedded pdb, and pdb as a file, correct? Can the tests be updated to check for both?

Reply:
We cannot test it for wasm because wasm debugger will not use this code path.

Reply:
not needed, it isn't a good symbol reading issue.